test/FunctionalTests/Microsoft.OData.Client.Tests/DataServiceContextHttpClientHandlerProviderTests.cs

```diff
@@ -34,6 +38,56 @@ public class DataServiceContextHttpClientHandlerProviderTests
 
         private const string PersonNameValue = "John Doe";
 
+        [Fact]
+        public void RdsExtendedTest()

```

Comment:
I'm not sure what's being tested here. How come there are no assertions? Or you could use a more descriptive name for the test.

Reply:
All very good points, I updated the name, adding a method summary, and added an extra branch in the test to illustrate the difference in expectation between the existing behavior and the new behavior.

I didn't think about it when I created the PR; I had written this test, which reproed the exception that the customer was getting, and then addressed the issue, causing this to pass (because no exception was thrown). I didn't think about the implementation any more after that...